### PR TITLE
Replace SettingsActivity with UiTable

### PR DESCRIPTION
**Pull request by singleNeuron**
Signed-off-by: singleNeuron <Cryolitia@gmail.com>

# Replace SettingsActivity with UiTable

## 更改内容 / Types of Changes

- 修复 Bugfix
- 新功能 New Feature
- 增强优化 Enhancement/optimization
- UI界面 User Interface
- 内核 Core
- 文档 Documentation
- 模板 Templates

## 由这个PR解决或关闭的Issues / Issues Fixed or Closed by This PR

-  

## 检查列表 / Checklist

- [x] 我的代码符合本仓库的代码规范 My code follows the code style of this project
- [x] 我已经测试了这些更改，它们可以正常工作，既不会破坏原有任何功能(或我可以解决)，也不会影响对旧版本的支持 I have tested the changes and verified that they work and don't break anything(as well as I can manage) or drop the support for previous versions.
- [x] 我已合并了对后续工作无意义的commit并确认不会对后续维护造成困扰 I have merged commits that are meaningless for follow-up work and confirmed that they will not cause damage to follow-up maintenance

### CODE REVIEW

File: app/src/main/java/cc/ioctl/hook/$endGiftHook.java [hunk 1]
```diff
@@ -24,24 +24,44 @@
...
-import nil.nadph.qnotified.hook.CommonDelayableHook;
 import nil.nadph.qnotified.step.DexDeobfStep;
 import nil.nadph.qnotified.util.DexKit;
 import nil.nadph.qnotified.util.Initiator;
 import nil.nadph.qnotified.util.LicenseStatus;
 
 @FunctionEntry
-public class $endGiftHook extends CommonDelayableHook {
+@UiItem
+public class $endGiftHook extends CommonDelayAbleHookBridge {
```
Comment (keta1) [inline app/src/main/java/cc/ioctl/hook/$endGiftHook.java:47]: 会丢配置文件
坏耶

File: app/src/main/java/me/singleneuron/qn_kernel/ui/base/UiRoutine.kt [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * and eula along with this software.  If not, see
+ * <https://www.gnu.org/licenses/>
+ * <https://github.com/ferredoxin/QNotified/blob/master/LICENSE.md>.
+ */
+
+package me.singleneuron.qn_kernel.ui.base
+
+val 净化功能 = arrayOf("净化功能")
```
Comment (keta1) [inline app/src/main/java/me/singleneuron/qn_kernel/ui/base/UiRoutine.kt:25]: 奇怪的中文变量

File: app/src/main/java/cc/ioctl/hook/打开资料卡.java [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+import android.widget.LinearLayout;
+
+import androidx.core.view.ViewCompat;
+
+import nil.nadph.qnotified.MainHook;
+import nil.nadph.qnotified.ui.CustomDialog;
+import nil.nadph.qnotified.ui.drawable.HighContrastBorder;
+import nil.nadph.qnotified.util.Toasts;
+
+public class 打开资料卡 {
```
Comment (keta1) [inline app/src/main/java/cc/ioctl/hook/打开资料卡.java:43]: 怎么还是中文

**Review by keta1 — CHANGES_REQUESTED** [review]
中文改了

**Comment by singleNeuron** [comment]
> 中文改了

不改

**Comment by keta1** [comment]
@singleNeuron

**Review by cinit — APPROVED** [review]

File: app/src/main/java/ltd/nextalone/hook/ChatWordsCount.kt [hunk 4]
```diff
@@ -34,18 +34,34 @@ import me.kyuubiran.util.getExFriendCfg
...
 import nil.nadph.qnotified.ui.ViewBuilder
 import nil.nadph.qnotified.util.Initiator
+import nil.nadph.qnotified.util.QQVersion
 import nil.nadph.qnotified.util.Toasts
 import nil.nadph.qnotified.util.Utils
 import java.util.*
 
 @FunctionEntry
-object ChatWordsCount : CommonDelayableHook("na_chat_words_count_kt") {
+@me.singleneuron.qn_kernel.annotation.UiItem
```
Comment (keta1) [inline app/src/main/java/ltd/nextalone/hook/ChatWordsCount.kt:52]: 怪

**Review by keta1 — APPROVED** [review]

**Pull request merged**